pkg/resource/deploy/step_generator.go

```diff
@@ -137,6 +146,61 @@ func (sg *stepGenerator) GenerateReadSteps(event ReadResourceEvent) ([]Step, res
 // If the given resource is a custom resource, the step generator will invoke Diff and Check on the
 // provider associated with that resource. If those fail, an error is returned.
 func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, result.Result) {
+	steps, res := sg.generateBasicSteps(event)
+	if res != nil {
+		contract.Assert(len(steps) == 0)
+		return nil, res
+	}
+
+	// we got a set of steps to perfom.  If any of the steps to perform depended on any resources we
+	// skipped creating (because they were not in a --target list) then issue an error that that
+	// create was necessary and that the user needed to supply it explicitly.
+	return sg.checkForSkippedCreateDependencies(steps)
+}
+
+func (sg *stepGenerator) checkForSkippedCreateDependencies(steps []Step) ([]Step, result.Result) {
+	for _, step := range steps {
+		// Note: we don't have an error if the step we're looking is itself another skipped-creation
+		// step (even if it references another skipped creation). Instead, we'll error out if *that*
+		// creation is used in an actual targeted change.
+		if sameStep, ok := step.(*SameStep); ok && sameStep.IsSkippedCreation() {
+			continue
+		}
+
+		new := step.New()
+		if new == nil {
+			continue
+		}
+
+		for _, urn := range new.Dependencies {
```

Comment:
@pgavlin we have `.Dependencies` and `.PropertyDependencies`.  Is it ok to read just ths one?

Reply:
Yes.

Reply:
In fact, you must read that one--it captures _all_ dependencies of the resource, including explicitly-specified dependencies that aren't present in `PropertyDependencies` (i.e. `dependsOn:`).